File: StructuralPattern/Decorator/Python/Concretes/Peashooter.py

from Interfaces import Plant

class Peashooter(Plant):
    """
        Peashooter class implementation for plant
    """
    def __init__(self, *args, **kwargs):
        self.name = kwargs.get('name', 'Peashooter')
        self.damage = kwargs.get('damage', 20)
        super().__init__(*args, **kwargs)

    def get_name(self) -> str:
        return self.name
    
    def set_name(self, name):
        self.name = name
    
    def attack(self) -> str:
        print(f'{self.name} attack dame: {self.damage}')

File: StructuralPattern/Decorator/Golang/Concretes/Peashooter.go
/**
 * Struct Peashooter implementation for Plant interface
 */

package c

import (
	"fmt"
)

type Peashooter struct {
	Name string
}

// Constructor
func NewPeashooter() *Peashooter {
	return &Peashooter{
		Name: "Peashooter",
	}
}

// Deloy method implementation
func (p *Peashooter) Attack() {
	fmt.Printf("%v is attacking\n", p.Name)
}

// Deloy method implementation
func (p *Peashooter) GetName() string {
	return p.Name
}
